onnxruntime/contrib_ops/cuda/quantization/matmul_nbits.cc

```diff
@@ -78,17 +77,18 @@
       SafeInt<int>(GetDeviceProp().sharedMemPerBlock),
       static_cast<cudaStream_t>(ctx->GetComputeStream()->GetHandle()));
   if (!is_4bit_done) {
-    int64_t K_padded = (K_ + block_size_ - 1) / block_size_ * block_size_;
```

Comment:
>int64_t K_padded = (K_ + block_size_ - 1) / block_size_ * block_size_; [](http://example.com/codeflow?start=4&length=70)

K_padded is for alignment. GPU memory is accessed in groups of 32 bits, 64 bits, and 128 bits. padded K can improve performance if K is not a multiplier of 8 for half type.

Reply:
Why do we care about performance when K is such strange number?

Reply:
When quantization block is horizontally oriented, this padding does not make sense. And do we really care about the perf of strange matrix shapes that rarely seen in any LLM?

Reply:
may try padding it to align to 8 for rowwise quantization. It is introduced for LLM. People may try it out for other models or scenarios.

Reply:
If the model author decides to use a strange shape, then the floating point GEMM perf would suffer because of this alignment issue already, why do we care?